gcovr/gcov_parser.py

```diff
@@ -276,6 +280,7 @@ def parse_coverage(
     logger: Logger,
     exclude_lines_by_pattern: Optional[str],
     flags: ParserFlags,
+    source_encoding: str,
```

Comment:
Source encoding is now unused. Removing this parameter requires updating the tests that call this function.